### PR TITLE
Added Google analytics data attribute

### PR DESCRIPTION
**Pull request by nehakannaujia**
@shaileshp0110  : Please review and merge.

### CODE REVIEW

**Comment by nehakannaujia** [comment]
@shaileshp0110  : Fixed Unit tests. Please review and merge.

**Review by shaileshp0110 — CHANGES_REQUESTED** [review]
Need to be optional field rather mandatory String

File: app/views/language_selection.scala.html [hunk 1]
```diff
@@ -2,15 +2,16 @@
...
 
 <p class="@if(customClass.isDefined) {@customClass.get}">
     @langMap.map { case (key: String, value: Lang) =>
             @if(lang.code != value.code) {
                 @Link.toInternalPage(
                     id      = Some(s"$key-switch"),
                     url     = s"${langToCall(key)}",
-                    value   = Some(key.capitalize)
+                    value   = Some(key.capitalize),
+                    dataAttributes = Some(Map("journey-click" -> s"${appName.fold("")(a=>a)}:language: ${value.code}"))
```
Comment (ryanb93) [inline app/views/language_selection.scala.html:14]: Use appName.getOrElse("") instead of fold.

File: test/views/LanguageSelectionSpec.scala [hunk 2]
```diff
@@ -35,50 +35,62 @@ class LanguageSelectionSpec @Inject()(val messagesApi: MessagesApi) extends Play
...
 
     "show a custom class if it is set" in running(new FakeApplication) {
-      val html = views.html.language_selection.render(languageMap, langToUrl(_), Some("float--right"), Welsh)
+      val html = views.html.language_selection.render(languageMap, langToUrl(_), Some("float--right"), None, Welsh)
       contentType(html) must be("text/html")
       contentAsString(html) must include("class=\"float--right\"")
     }
 
+    "show a data-journey-click attribute for GA if it is set and language is Welsh" in running(new FakeApplication) {
+      val html = views.html.language_selection.render(languageMap, langToUrl(_), None, Some("appName:language: cy-GB"), Welsh)
```
Comment (ryanb93) [inline test/views/LanguageSelectionSpec.scala:72]: Shouldn't the Some here just contain the app name by itself, then the test should check that Welsh was correctly injected into the resulting string. At the moment I think that the test passes but is a little misleading, the full string would be:

    "data-journey-click=\"appName:language: cy-GB\"":language: cy-GB\""

Based on the logic of:

    s"${appName.fold("")(a=>a)}:language: ${value.code}"

Comment (nehakannaujia) [reply test/views/LanguageSelectionSpec.scala:72]: @ryanb93  Yeah that's correct. It should be only "appName" instead of complete string.

**Pull request merged**